x-pack/plugins/security_solution/public/resolver/view/use_camera.test.tsx

```diff
@@ -22,45 +24,120 @@ import { createStore } from 'redux';
 import { resolverReducer } from '../store/reducer';
 import { mockTreeFetcherParameters } from '../mocks/tree_fetcher_parameters';
 import { entityIDSafeVersion } from '../../../common/endpoint/models/event';
+import { act } from 'react-dom/test-utils';
 
 describe('useCamera on an unpainted element', () => {
-  let element: HTMLElement;
+  /** Enzyme full DOM wrapper for the element the camera is attached to. */
+  let element: ReactWrapper;
+  /**
+   * Enzyme full DOM wrapper for the alternate element that the camera can be attached to. Used for testing that the `ResizeObserver` attaches itself to the latest `ref`.
+   */
+  let alternateElement: ReactWrapper;
+  /**
+   * projection matrix returned by camera on last render.
+   */
   let projectionMatrix: Matrix3;
+  /**
+   * A `data-test-subj` ID used to identify the element the camera normally attaches to.
+   */
   const testID = 'camera';
-  let reactRenderResult: RenderResult;
+  /**
+   * A `data-test-subj` ID used to identify the element the camera alternatively attaches to.
+   */
+  const alternateTestID = 'alternate';
+  /**
+   * Returned by the legacy framework's render/mount function.
+   */
+  let wrapper: ReactWrapper;
   let store: ResolverStore;
   let simulator: SideEffectSimulator;
 
-  beforeEach(async () => {
-    store = createStore(resolverReducer);
-
-    const Test = function () {
-      const camera = useCamera();
-      const { ref, onMouseDown } = camera;
-      projectionMatrix = camera.projectionMatrix;
-      return <div data-test-subj={testID} onMouseDown={onMouseDown} ref={ref} />;
-    };
+  /** Used to find an element by the data-test-subj attribute.
+   */
+  let domElementByTestSubj: (testSubj: string) => ReactWrapper;
 
-    simulator = sideEffectSimulatorFactory();
+  /**
+   * Yield the result of `mapper` over and over, once per event-loop cycle.
+   * After 10 times, quit.
+   * Use this to continually check a value. See `toYieldEqualTo`.
+   */
+  async function* map<R>(mapper: () => R): AsyncIterable<R> {
+    let timeoutCount = 0;
+    while (timeoutCount < 10) {
+      timeoutCount++;
+      yield mapper();
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          wrapper.update();
+          resolve();
+        }, 0);
+      });
+    }
+  }
 
-    reactRenderResult = render(
-      <Provider store={store}>
-        <SideEffectContext.Provider value={simulator.mock}>
-          <Test />
+  function TestWrapper({
```

Comment:
Use a top level wrapper component so we can call `setProps`. We use this to switch the DOM element the camera is attached to. TODO, add this comment to code.